install/infra/modules/eks/storage.tf

```diff
@@ -0,0 +1,66 @@
+resource "aws_s3_bucket" "gitpod-storage" {
+  count = var.enable_external_storage ? 1 : 0
+
+  force_destroy = true
+  bucket        = "bucket-${var.cluster_name}"
+  acl           = "private"
+}
+
+resource "aws_s3_bucket" "gitpod-registry-backend" {
+  count = var.enable_external_storage_for_registry_backend ? 1 : 0
+
+  force_destroy = true
+  bucket        = "reg-bucket-${var.cluster_name}"
+  acl           = "private"
+}
+
+resource "aws_s3_bucket_versioning" "storage" {
+  count = var.enable_external_storage ? 1 : 0
+
+  bucket = aws_s3_bucket.gitpod-storage[0].id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_versioning" "registry" {
+  count = var.enable_external_storage_for_registry_backend ? 1 : 0
+
+  bucket = aws_s3_bucket.gitpod-registry-backend[0].id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+data "aws_iam_policy_document" "s3_policy" {
+  count = var.enable_external_storage ? 1 : 0
+  statement {
+    actions   = ["s3:*"]
```

Comment:
**Recommendation:** This permission set is very broad, can we restrict this to the permission set specified in the [Object Storage](https://www.gitpod.io/docs/self-hosted/latest/reference-architecture/single-cluster-ref-arch#object-storage) section? (We may need to have separate policy documents for the object storage bucket and registry bucket if they're both defined.)

Reply:
I have updated this! I think the storage file has ended up becoming a bit convoluted, but I will clean it up in the future